symengine/sets.h

```diff
@@ -199,5 +225,42 @@ inline RCP<const Set> interval(const RCP<const Number> &start,
         return finiteset({start});
     return emptyset();
 }
+
+// ! \return RCP<const Set>
+inline RCP<const Set> sym_union(const set_basic &in, bool solve = true)
+{
+    set_basic input;
+    if (solve == false && in.size() > 1)
+        return make_rcp<const Union>(in);
+    set_basic combined_FiniteSet;
+    for (auto it = in.begin(); it != in.end(); ++it) {
+        if (is_a<FiniteSet>(**it)) {
+            set_basic container;
+            const FiniteSet &other = static_cast<const FiniteSet &>(**it);
+            std::set_union(combined_FiniteSet.begin(), combined_FiniteSet.end(),
+                           other.container_.begin(), other.container_.end(),
+                           std::inserter(container, container.begin()),
+                           RCPBasicKeyLess{});
+            combined_FiniteSet = container;
```

Comment:
`combined_FiniteSet.insert(other.container_.begin(), other.container_.end())` is better here.